ui/raidboss/timeline_parser.ts

```diff
@@ -185,6 +185,8 @@ export class TimelineParser {
   private labelToTime: { [name: string]: number } = {};
   // Map of encountered syncs to the label they are jumping to.
   private labelToSync: { [name: string]: Sync[] } = {};
+  // Used in subclass (update_logdefs), but defined here due to being utilized during construction
+  public entries: Partial<Record<LogDefinitionName, number[]>> | undefined;
```

Comment:
Could this exist on the subtype instead, and the type assertion be changed to account for it? e.g.
```ts
type TimelineParserWithEntries = TimelineParser & {
  entries: Partial<Record<LogDefinitionName, number[]>> | undefined;
};
// ...
    const triggerSet = (await import(importPath)).default as TimelineParserWithEntries;
```

(feel free to name the type something else)

Or you could put the `entries` property on the subclass directly and just `.default as TimelineTypeExtractor`?

Reply:
I don't think this would quite work, as timelines aren't imported - `TimelineParser` takes a string input (file read with newlines) and the constructor calls `parse()` to process the string.  Or am I misunderstanding your suggestion?

Taking a step back though, the reason I added `entries` as a property on the parent instead of the subclass was because the parent constructor calls `parse()` (which starts the method chain that ultimately results in `entries` being populated).  So defining it on the subclass results in re-initialization that wipes the data, as subclass props are initialized after parent class construction.  It was a bit too tortured to try to handle type extraction as its own callable method outside of the `parse()` chain, because all of the regex/entry processing is done by that point, and I didn't want to change `parse()` to be a callable method, since that would require reworking all current implementations of `TimelineParser`.  I wasn't crazy about making `entries` a parent prop, but it was the simplest solution and at least made logical sense, since the parent class is technically populating it.

Reply:
This diff works fine for me, no errors running the npm script and makes the expected changes to the netlog defs ts file.

```patch
diff --git a/ui/raidboss/timeline_parser.ts b/ui/raidboss/timeline_parser.ts
index 23b08d18b..0375f0658 100644
--- a/ui/raidboss/timeline_parser.ts
+++ b/ui/raidboss/timeline_parser.ts
@@ -185,8 +185,6 @@ export class TimelineParser {
   private labelToTime: { [name: string]: number } = {};
   // Map of encountered syncs to the label they are jumping to.
   private labelToSync: { [name: string]: Sync[] } = {};
-  // Used in subclass (update_logdefs), but defined here due to being utilized during construction
-  public entries: Partial<Record<LogDefinitionName, number[]>> | undefined;
 
   constructor(
     text: string,
diff --git a/util/update_logdefs.ts b/util/update_logdefs.ts
index d3ef2ed36..5437619b3 100644
--- a/util/update_logdefs.ts
+++ b/util/update_logdefs.ts
@@ -47,6 +47,8 @@ type FileMatch = {
 type FileMatches = Partial<Record<LogDefinitionName, FileMatch[]>>;
 
 class TimelineTypeExtractor extends TimelineParser {
+  public entries: Partial<Record<LogDefinitionName, number[]>> = {};
+
   constructor(contents: string) {
     super(contents, [], []);
   }

```

Reply:
> This diff works fine for me, no errors running the npm script and makes the expected changes to the netlog defs ts file.

It makes the same changes to `netlog_defs` right now because all of the new log types being added to the analysis filter can be and are being picked up from trigger file parsing; the timeline parsing has some log-type overlap but isn't adding any unique log types by itself, so comparing `netlog_defs` won't give the full picture of whether timeline parsing is working correctly.  OTOH, if you compare the script console outputs from having `entries` on the parent class vs. subclass, you'll see that having it on the subclass isn't picking up any of the uses of these log types in timeline files (see attached output).  (fwiw, i confirmed this with some debug console logging that showed that `entries` was empty after the subclass completed initialization).
[_parentprop_stdout.txt](https://github.com/OverlayPlugin/cactbot/files/15102914/_parentprop_stdout.txt)
[_subclassprop_stdout.txt](https://github.com/OverlayPlugin/cactbot/files/15102916/_subclassprop_stdout.txt)

Reply:
Okay, I see the confusion now. I think for now, update the comment to include a `TODO` to revisit after refactoring the timeline parser code, and later on we can make `parse` public and move responsibility for calling `parse` up to the code that's constructing the `TimelineParser`?

Reply:
> Okay, I see the confusion now. I think for now, update the comment to include a `TODO` to revisit after refactoring the timeline parser code, and later on we can make `parse` public and move responsibility for calling `parse` up to the code that's constructing the `TimelineParser`?

Actually, I think I might've figured out a way to do this that will move the prop to the subclass.  I can add an optional boolean param to `TimelineParser` that controls the execution of `parse()` in the constructor, make `parse()` public, and then have the subclass initialize the parent without calling `parse()`, and just manually call it in the update script after the subclass is fully initialized.  Not quite as clean, but it shouldn't affect any current implementations.  Although I don't know if making `parse()` public in and of itself will cause any runtime degradation?

Reply:
> Why not just call `parse` directly as part of the subclass's constructor? e.g.

Hey, even better.